backend/groth16/bls12-377/verify.go

```diff
@@ -87,8 +93,8 @@ func Verify(proof *Proof, vk *VerifyingKey, publicWitness fr.Vector) error {
 	}
 	kSum.AddMixed(&vk.G1.K[0])
 
-	if vk.HasCommitment {
-		kSum.AddMixed(&proof.Commitment)
+	for i := range proof.Commitments {
```

Comment:
same, my understanding is we probably won't go over comuple of commitments, but if we do, this place might be a good candidate for a batchAddition